common/models/templates/src/trie.ts

```diff
@@ -280,13 +291,14 @@ export class Trie {
    */
   toKey: Wordform2Key;
 
-  constructor(root: Node, totalWeight: number, wordform2key: Wordform2Key) {
+  constructor(trie: Node | string, totalWeight: number, wordform2key: Wordform2Key) {
+    const root = (typeof trie == 'string') ? decompressNode(trie, wordform2key, 0) : trie;
     this.root = root;
```

Comment:
```suggestion
    this.root = (typeof trie == 'string') ? decompressNode(trie, wordform2key, 0) : trie;
```

Seems to be a lot of redundant variable declarations?